src/System.Management.Automation/CoreCLR/CorePsAssemblyLoadContext.cs

```diff
@@ -587,4 +587,28 @@ public static void SetPowerShellAssemblyLoadContext([MarshalAs(UnmanagedType.LPW
             PowerShellAssemblyLoadContext.InitializeSingleton(basePaths);
         }
     }
+
+    /// <summary>
+    /// Provides helper functions to faciliate calling managed code from a native PowerShell host.
+    /// </summary>
+    public static class PowerShellUnsafeAssemblyLoad
+    {
+        /// <summary>
+        /// Load an assembly in memory from unmanaged code.
+        /// </summary>
+        /// <param name="data">
+        /// Unmanaged pointer to assembly data buffer.
+        /// </param>
+        /// <param name="size">
+        /// Size in bytes of the assembly data buffer.
+        /// </param>
+        [UnmanagedCallersOnly]
+        public static void LoadAssemblyFromMemory(IntPtr data, int size)
+        {
+            byte[] bytes = new byte[size];
+            Marshal.Copy(data, bytes, 0, size);
+            using var stream = new MemoryStream(bytes);
+            AssemblyLoadContext.Default.LoadFromStream(stream);
```

Comment:
Can we use UnmanagedMemoryStream to avoid copying three times?
https://docs.microsoft.com/en-us/dotnet/api/system.io.unmanagedmemorystream?view=net-5.0

Also it seems it will does not work on some ARM CPUs. If so we should add this in remark comments.

Reply:
I just pushed an update to use UnmanagedMemoryStream instead of MemoryStream and marked the class as unsafe in order to use it. I guess it's fine since the class is now called PowerShellUnsafeAssemblyLoad. I tested it and it still works just fine.

Do you have more details about the possible problem with ARM CPUs? I couldn't find a mention of it anywhere, and I'm not sure what the problem might be. I know some ARM CPUs are quite picky with regards to memory alignment, but this is a byte array, so it shouldn't be much of a problem, is it?

Reply:
> @iSazonov just to make sure, the "int" type in .NET is always a signed 32-bit integer, right? That would have been the only thing left if it wasn't the case, so I'm just double checking. I only care that it has a predictable size, and there's no point in making it 64-bit really.

Yes, it is 32-bit. And it is _blittable_ type - such types can be passed between managed and unmanaged code without marshalling https://docs.microsoft.com/en-us/dotnet/framework/interop/blittable-and-non-blittable-types

Reply:
No, it's just that there are sometimes surprises with ARMs. You will only know about this if you check your host on them. :-)